### PR TITLE
Migrate IPF code from caf.toolkit

### PR DESCRIPTION
**Pull request by BenTaylor-TfN**
Describe Changes
---
Code being removed from caf.toolkit can be seen here:
https://github.com/Transport-for-the-North/caf.toolkit/pull/105

Task Checklist
----
- [ ] Have unittests been added (testing individual functions and their edge cases)
- [ ] Have integration tests been added (if applicable, to test modules of functionality)
- [ ] Updated RELEASE.md to reflect changes in PR
- [ ] Have new dependencies been added?
  - [ ] Have they been added to either `requirements.txt` or `requirements_dev.txt`.
  - [ ] Have they been added to `setup.cfg`

### CODE REVIEW

File: src/caf/distribute/array_utils.py [hunk 1]
```diff
@@ -0,0 +1,538 @@
+# -*- coding: utf-8 -*-
```
Comment (isaac-tfn) [inline src/caf/distribute/array_utils.py:1]: I think any references to array_utils within this package need updating. This probably still passes tests as current released versions of toolkit still contain what's needed.

File: src/caf/distribute/iterative_proportional_fitting.py [hunk 2]
```diff
@@ -0,0 +1,995 @@
+# -*- coding: utf-8 -*-
```
Comment (isaac-tfn) [inline src/caf/distribute/iterative_proportional_fitting.py:1]: Is this ported as is from toolkit?

**Review by isaac-tfn — COMMENTED** [review]
Couple of questions - will review more fully on Monday!
